### PR TITLE
Do not ignore CA when downstream skip verify is true

### PR DESCRIPTION
**Pull request by sunjayBhatia**
- if no CA set and skip verify is true, client certs are not required by
Envoy
- if CA set and skip verify is true, client certs are required by Envoy
- CA still required if skip verify is false

### CODE REVIEW

**Comment by skriss** [comment]
This is probably worth some updated documentation, maybe in the API godoc for these fields and also in https://projectcontour.io/docs/main/config/tls-termination/#client-certificate-validation, since the behavior here might not be the most intuitive

**Review by youngnick — CHANGES_REQUESTED** [review]
The code looks good to me, but I think the documentation updates are a must-have.

**Review by youngnick — APPROVED** [review]
Doc updates look good, 👍

**Comment by skriss** [comment]
You've got a merge conflict to resolve now

**Review by skriss — APPROVED** [review]

**Comment by sunjayBhatia** [comment]
hold for merging until https://github.com/projectcontour/contour/pull/3665 is done, want to add to the integration test that is int that PR

**Comment by stevesloka** [comment]
@sunjayBhatia looks like #3665 has merged so this is good to go. Also needs a rebase. =)

**Comment by sunjayBhatia** [comment]
still having an issue with an integration test that I just added, not sure why the "accept untrusted" setting with a CA is not passing on a request that comes with certs from a different CA

File: test/e2e/httpproxy/007_client_cert_auth_test.go [hunk 1]
```diff
@@ -405,24 +474,31 @@ func testClientCertAuth(fx *e2e.Framework) {
...
 }
 
-func optUseCert(cert tls.Certificate) func(*tls.Config) {
+func optUseClientCert(cert *tls.Certificate) func(*tls.Config) {
 	return func(c *tls.Config) {
-		c.Certificates = append(c.Certificates, cert)
+		// Use c.GetClientCertificate rather than setting c.Certificates so the
+		// client cert specified is always presented, regardless of the request
+		// details from the server.
+		c.GetClientCertificate = func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
```
Comment (sunjayBhatia) [inline test/e2e/httpproxy/007_client_cert_auth_test.go:499]: note this change, just commenting for visibility

File: test/e2e/httpproxy/007_client_cert_auth_test.go [hunk 2]
```diff
@@ -405,24 +474,31 @@ func testClientCertAuth(fx *e2e.Framework) {
...
+			return cert, nil
+		}
 	}
 }
```
Comment (skriss) [inline test/e2e/httpproxy/007_client_cert_auth_test.go:503]: aha, 👍 figuring that out

**Review by skriss — APPROVED** [review]

**Comment by skriss** [comment]
@sunjayBhatia good to merge here?

**Pull request merged**